syntax = "proto3";

service ReservationCrud {
  rpc GetAll (Empty) returns (ReservationDtos);
  rpc GetById (ReservationId) returns (ReservationDto);
  rpc Create (ReservationDto) returns (Result);
  rpc Delete (ReservationId) returns (Result);
  rpc Update (ReservationDto) returns (Result);
  rpc GetByHost(HostId) returns (ReservationDtos);
  rpc GetByGuest(GuestId) returns (ReservationDtos);
  rpc GetReservationsForAcceptance(ReservationDto) returns (ReservationDtos);
  rpc AcceptReservation (ReservationDto) returns (Result);
  rpc GetPendingReservationsByHost(HostId) returns (ReservationDtos);
  rpc GetActiveByGuest(GuestId) returns (ReservationDtos);
  rpc GetActiveByHost(HostId) returns (ReservationDtos);
}

message Empty {
}


message Result{
    string status = 1;
}
message ReservationDto {
    string reservation_id = 1;
    string accommodation_id = 2;
    string host_id = 3;
    Guest guest = 4;
    int32 number_of_guests = 5 ;
    string beginning_date = 6;
    string ending_date = 7;
    double total_price = 8;
    int32 status = 9;
}

message ReservationDtos {
  repeated ReservationDto items = 1;
}

message ReservationId {
  string id = 1;
}
message HostId {
  string id = 1;
}
message Guest {
  string guest_id = 1;
  int32 canceledReservations = 2;
}
message GuestId{
  string guest_id = 1;
}